File: iosApp/iosApp/view/BiographyInfo.swift
import SwiftUI
import Shared

struct BiographyInfo: View {
    let biography: BiographyResponse

    var body: some View {
        let employedFromDate = parseDate(from: biography.employedFrom)
        let employedFromFormatted = formatDate(employedFromDate)

        VStack(alignment: .leading, spacing: 8) {
            Group {
                LabelView(label: "Titul", value: biography.title ?? "")
                LabelView(label: "Jméno", value: biography.firstName)
                LabelView(label: "Příjmení", value: biography.lastName)
                LabelView(label: "E-Mail", value: biography.email)
                LabelView(label: "Telefon", value: biography.phone)
            }

            Divider().padding(.vertical, 8)

            Group {
                LabelView(label: "Ulice", value: biography.street)
                LabelView(label: "Město", value: biography.city)
                LabelView(label: "Stát", value: biography.country)
            }

            Divider().padding(.vertical, 8)

            Group {
                LabelView(label: "Pracovní pozice", value: biography.position)
                LabelView(label: "Zaměstnán/a od", value: employedFromFormatted)
            }
        }
    }

    private func parseDate(from string: String) -> Date? {
        let formatter = DateFormatter()
        formatter.dateFormat = "yyyy-MM-dd"
        return formatter.date(from: string)
    }

    private func formatDate(_ date: Date?) -> String {
        guard let date = date else { return "-" }
        let formatter = DateFormatter()
        formatter.dateFormat = "dd.MM.yyyy"
        return formatter.string(from: date)
    }
}

struct LabelView: View {
    let label: String
    let value: String

    var body: some View {
        VStack(alignment: .leading, spacing: 2) {
            Text(label)
                .font(.system(size: 14, weight: .bold))
                .foregroundColor(.black)
            Text(value)
                .font(.system(size: 14))
                .foregroundColor(.black)
        }
    }
}

File: composeApp/src/androidMain/kotlin/org/adastra/curriculum/view/BiographyInfo.kt
package org.adastra.curriculum.view

import androidx.compose.foundation.layout.padding
import androidx.compose.material.TabRowDefaults.Divider
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import kotlinx.datetime.LocalDate
import org.adastra.curriculum.client.data.BiographyResponse

@Composable
fun BiographyInfo(biography: BiographyResponse) {
    val employedFrom = LocalDate.parse(biography.employedFrom)
    val employedFromFormatted = "%02d.%02d.%04d".format(
        employedFrom.dayOfMonth,
        employedFrom.monthNumber,
        employedFrom.year
    )

    Text(
        "Titul",
        fontSize = 14.sp,
        fontWeight = FontWeight.Bold,
        color = Color.Black
    )
    Text(
        biography.title ?: "",
        fontSize = 14.sp,
        fontWeight = FontWeight.Normal,
        color = Color.Black
    )
    Text(
        "Jméno",
        fontSize = 14.sp,
        fontWeight = FontWeight.Bold,
        color = Color.Black
    )
    Text(
        biography.firstName,
        fontSize = 14.sp,
        fontWeight = FontWeight.Normal,
        color = Color.Black
    )
    Text(
        "Příjmení",
        fontSize = 14.sp,
        fontWeight = FontWeight.Bold,
        color = Color.Black
    )
    Text(
        biography.lastName,
        fontSize = 14.sp,
        fontWeight = FontWeight.Normal,
        color = Color.Black
    )
    Text(
        "E-Mail",
        fontSize = 14.sp,
        fontWeight = FontWeight.Bold,
        color = Color.Black
    )
    Text(
        biography.email,
        fontSize = 14.sp,
        fontWeight = FontWeight.Normal,
        color = Color.Black
    )
    Text(
        "Telefon",
        fontSize = 14.sp,
        fontWeight = FontWeight.Bold,
        color = Color.Black
    )
    Text(
        biography.phone,
        fontSize = 14.sp,
        fontWeight = FontWeight.Normal,
        color = Color.Black
    )
    Divider(
        color = Color.Gray,
        thickness = 1.dp,
        modifier = Modifier.padding(vertical = 8.dp)
    )
    Text(
        "Ulice",
        fontSize = 14.sp,
        fontWeight = FontWeight.Bold,
        color = Color.Black
    )
    Text(
        biography.street,
        fontSize = 14.sp,
        fontWeight = FontWeight.Normal,
        color = Color.Black
    )
    Text(
        "Město",
        fontSize = 14.sp,
        fontWeight = FontWeight.Bold,
        color = Color.Black
    )
    Text(
        biography.city,
        fontSize = 14.sp,
        fontWeight = FontWeight.Normal,
        color = Color.Black
    )
    Text(
        "Stát",
        fontSize = 14.sp,
        fontWeight = FontWeight.Bold,
        color = Color.Black
    )
    Text(
        biography.country,
        fontSize = 14.sp,
        fontWeight = FontWeight.Normal,
        color = Color.Black
    )
    Divider(
        color = Color.Gray,
        thickness = 1.dp,
        modifier = Modifier.padding(vertical = 8.dp)
    )
    Text(
        "Pracovní pozice",
        fontSize = 14.sp,
        fontWeight = FontWeight.Bold,
        color = Color.Black
    )
    Text(
        biography.position,
        fontSize = 14.sp,
        fontWeight = FontWeight.Normal,
        color = Color.Black
    )
    Text(
        "Zaměstnán/a od",
        fontSize = 14.sp,
        fontWeight = FontWeight.Bold,
        color = Color.Black
    )
    Text(
        employedFromFormatted,
        fontSize = 14.sp,
        fontWeight = FontWeight.Normal,
        color = Color.Black
    )
}